main.go

```diff
@@ -280,54 +278,69 @@ func (s *Session) firstNav(ctx context.Context) error {
 	return nil
 }
 
-// navToEnd waits for the page to be ready to receive scroll key events, by
-// trying to select an item with the right arrow key, and then scrolls down to the
-// end of the page, i.e. to the oldest items.
+// navToEnd waits for the page to be ready to receive scroll key events,
+// and then scrolls down to the end of the page, i.e. to the oldest items.
 func navToEnd(ctx context.Context) error {
-	// wait for page to be loaded, i.e. that we can make an element active by using
-	// the right arrow key.
-	for {
-		chromedp.KeyEvent(kb.ArrowRight).Do(ctx)
-		time.Sleep(tick)
-		var ids []cdp.NodeID
-		if err := chromedp.Run(ctx,
-			chromedp.NodeIDs(`document.activeElement`, &ids, chromedp.ByJSPath)); err != nil {
-			return err
-		}
-		if len(ids) > 0 {
-			if *verboseFlag {
-				log.Printf("We are ready, because element %v is selected", ids[0])
-			}
-			break
-		}
-		time.Sleep(tick)
-	}
+	// wait for page to be loaded
+	chromedp.WaitReady("body", chromedp.ByQuery).Do(ctx)
 
-	// try jumping to the end of the page. detect we are there and have stopped
-	// moving when two consecutive screenshots are identical.
-	var previousScr, scr []byte
+	// Try scrolling to the end of the page.
+	// After each scroll attempt we extract the last Photo Page Element (href) from the DOM.
+	// We detect we are at the end when two consecutive DOM extractions are identical.
+	var previousHref string
 	for {
 		chromedp.KeyEvent(kb.PageDown).Do(ctx)
 		chromedp.KeyEvent(kb.End).Do(ctx)
-		chromedp.CaptureScreenshot(&scr).Do(ctx)
-		if previousScr == nil {
-			previousScr = scr
-			continue
+		time.Sleep(tick) // sleep *before* we establish new state in DOM
+
+		// Extract last Photo Page Element (href) from DOM.
+		lastHrefInDOM, err := lastPhotoInDOM(ctx)
+		if err != nil {
+			continue // Just ignore this error, continue will retry.
 		}
-		if bytes.Equal(previousScr, scr) {
+		if previousHref == lastHrefInDOM {
 			break
 		}
-		previousScr = scr
-		time.Sleep(tick)
+		previousHref = lastHrefInDOM
 	}
 
-	if *verboseFlag {
-		log.Printf("Successfully jumped to the end")
+	// Now that we have stopped scrolling, select (focus) on the last element
+	// The element must be focused, so that navToLast can send "\n" to enter photo detail page
+	lastEltSel := fmt.Sprintf(`a[href="%s"]`, previousHref)
+	if err := chromedp.Focus(lastEltSel).Do(ctx); err != nil {
+		return err
 	}
 
+	if *verboseFlag {
+		log.Printf("Successfully jumped to the end: %s", previousHref)
```

Comment:
Hmm, then I don't think we should have previousHref in this logging message, as it might lead one to believe that previousHref is the very last element of the page (which it might not be, as you've just explained).